cores/arduino/stm32/twi.c

```diff
@@ -367,6 +367,11 @@ void i2c_slave_write_IT(i2c_t *obj, uint8_t *data, uint8_t size)
 {
   uint8_t i = 0;
 
+  // Protection to not override the TxBuffer
+  if(size > I2C_TXRX_BUFFER_SIZE) {
+    size = I2C_TXRX_BUFFER_SIZE;
+  }
+
```

Comment:
Shouldn't we rather return en error?
Otherwise application would get an incomplete frame and will not get an error ...

Reply:
I quickly patched a possible memory leak so I kept the function "void". 
I can return 0 in case of error if you rather.

Reply:
What I mean is that you need to properly manage errors.

Reply:
Done!